component/azstorage/config.go

```diff
@@ -300,10 +317,10 @@ func ParseAndValidateConfig(az *AzStorage, opt AzStorageOptions) error {
 		}
 	}
 	az.stConfig.authConfig.Endpoint = opt.Endpoint
-	az.stConfig.authConfig.Endpoint = formatEndPoint(az.stConfig.authConfig.Endpoint, opt.UseHTTP)
+	az.stConfig.authConfig.Endpoint = formatEndpointProtocol(az.stConfig.authConfig.Endpoint, opt.UseHTTP)
+	az.stConfig.authConfig.Endpoint = formatEndpointAccountType(az.stConfig.authConfig.Endpoint, az.stConfig.authConfig.AccountType)
 
 	az.stConfig.authConfig.ActiveDirectoryEndpoint = opt.ActiveDirectoryEndpoint
-	az.stConfig.authConfig.ActiveDirectoryEndpoint = formatEndPoint(az.stConfig.authConfig.ActiveDirectoryEndpoint, false)
```

Comment:
Lets keep this, if we look closely we check if the string contains :// or not, if its not there we add it based on selected protocol, if its there we do not perform any transformations. So in case of AAD endpoint if user has given fully qualified URL it will not touch it, if its not given then anyway we are not sure whether to take http or https path and based on selected protocol we will tweak it which shall be ok.

Reply:
Or other way to look at this is as user has not given fully qualified URI, it may not work and we can ask user to correct his input. if we wish to take that path then removing this link is fine.